pkg/trace/sampler/remote_rates.go

```diff
@@ -0,0 +1,218 @@
+package sampler
+
+import (
+	"sync"
+	"time"
+
+	"github.com/DataDog/datadog-agent/pkg/config/remote/service"
+	"github.com/DataDog/datadog-agent/pkg/proto/pbgo"
+	"github.com/DataDog/datadog-agent/pkg/trace/config"
+	"github.com/DataDog/datadog-agent/pkg/trace/pb"
+	"github.com/DataDog/datadog-agent/pkg/trace/watchdog"
+	"github.com/DataDog/datadog-agent/pkg/util/log"
+)
+
+// RemoteRates sharded by signature. Allowing an independant feedback
+// loop per signature. RemoteRates adjusts sampling rates communicated
+// to the agent based on the observed traffic. It targets a TPS configured
+// remotely.
+type RemoteRates struct {
+	samplers   map[Signature]*Sampler
```

Comment:
We should track the size of this map. Or, even better, track the number of `Signature`s. We should be able to monitor how large this map gets.

Reply:
added
```
metrics.Gauge("datadog.trace_agent.remote.samplers", float64(len(r.samplers)), nil, 1)
metrics.Gauge("datadog.trace_agent.remote.sig_targets", float64(len(r.tpsTargets)), nil, 1)
```